system-test/storage.js

```diff
@@ -28,63 +26,57 @@ var through = require('through2');
 var tmp = require('tmp');
 var uuid = require('node-uuid');
 
+var Bucket = require('../lib/storage/bucket.js');
 var env = require('./env.js');
-var gcloud = require('../lib')(env);
-var storage = gcloud.storage();
-
-var BUCKET_NAME = generateBucketName();
-
-var files = {
-  logo: {
-    path: 'system-test/data/CloudPlatform_128px_Retina.png'
-  },
-  big: {
-    path: 'system-test/data/three-mb-file.tif'
-  },
-  html: {
-    path: 'system-test/data/long-html-file.html'
-  },
-  gzip: {
-    path: 'system-test/data/long-html-file.html.gz'
-  }
-};
-
-function deleteFile(file, callback) {
-  file.delete(callback);
-}
-
-function writeToFile(file, contents, callback) {
-  var writeStream = file.createWriteStream();
-  writeStream.once('error', callback);
-  writeStream.once('finish', callback.bind(null, null));
-  writeStream.end(contents);
-}
-
-function generateBucketName() {
-  return 'gcloud-test-bucket-temp-' + uuid.v1();
-}
-
-function setHash(obj, file, done) {
-  var hash = crypto.createHash('md5');
-  fs.createReadStream(obj[file].path)
-    .on('data', hash.update.bind(hash))
-    .on('end', function() {
-      obj[file].hash = hash.digest('base64');
-      done();
-    });
-}
+var File = require('../lib/storage/file.js');
+var Storage = require('../lib/storage/index.js');
 
 describe('storage', function() {
-  var bucket = storage.bucket(BUCKET_NAME);
+  var TESTS_PREFIX = 'gcloud-test-';
+
+  var storage = new Storage(env);
+  var bucket = storage.bucket(generateName());
+
+  var FILES = {
+    logo: {
+      path: 'system-test/data/CloudPlatform_128px_Retina.png'
+    },
+    big: {
+      path: 'system-test/data/three-mb-file.tif'
+    },
+    html: {
+      path: 'system-test/data/long-html-file.html'
+    },
+    gzip: {
+      path: 'system-test/data/long-html-file.html.gz'
+    }
+  };
 
   before(function(done) {
     bucket.create(done);
   });
 
   after(function(done) {
```

Comment:
@jgeewax we've run into an issue with Travis, any chance you could fix that for us? Or just add us to the project so we don't have to ask you to fix it for us :stuck_out_tongue_winking_eye: